src/System.Private.ServiceModel/tools/test/Bridge/Bridge/Program.cs

```diff
@@ -253,56 +284,90 @@ private static void ResetBridge(CommandLineArguments commandLineArgs)
+                if (!bridgeRunning)
+                {
+                    Console.WriteLine("The Bridge.exe process is not running so it will be started now.");
+                    break;
+                }
             }
 
             Process bridgeProcess = StartBridgeInNewProcess(commandLineArgs);
-            DateTime startTime = DateTime.Now;
+  
             while (((DateTime.Now - startTime).TotalSeconds) < commandLineArgs.RequireBridgeTimeoutSeconds)
             {
                 if (bridgeProcess.HasExited)
                 {
                     int exitCode = bridgeProcess.ExitCode;
                     Console.WriteLine("The Bridge process terminated unexpectedly with exit code {0}", bridgeProcess.ExitCode);
-                    Environment.Exit(-1);
+                    return Bridge_ExitCode_NotRunning;
                 }
 
                 if (PingBridge(commandLineArgs.BridgeConfiguration.BridgeHost,
                                commandLineArgs.BridgeConfiguration.BridgePort,
-                               out errorMessage))
+                               out errorMessage,
+                               out properties))
                 {
                     Console.WriteLine("The Bridge has been started successfully.");
-                    Environment.Exit(0);
+                    return Bridge_ExitCode_Started;
                 }
 
+                Console.WriteLine("  -- the ping failure response was: {0}", errorMessage);
                 Thread.Sleep(1000);
             }
 
             Console.WriteLine("The Bridge did not respond in the required {0} seconds.", 
                               commandLineArgs.RequireBridgeTimeoutSeconds);
-            Environment.Exit(-1);
+            return Bridge_ExitCode_NotRunning;
         }
 
 
         // Starts the Bridge locally in a new process
         private static Process StartBridgeInNewProcess(CommandLineArguments commandLineArgs)
         {
-            // Pass through the original command line arguments except for the -require switch
-            HashSet<string> originalArgHashSet = new HashSet<string>(commandLineArgs.OriginalArgs, StringComparer.OrdinalIgnoreCase);
-            originalArgHashSet.Remove("-require");
-            originalArgHashSet.Remove("/require");
+            // Pass through the original command line arguments except for the -require switch.
+            // Otherwise we would recursively start this same logic in a new process.
+            List<string> originalArgList = new List<string>(commandLineArgs.OriginalArgs);
+            for (int i = originalArgList.Count-1; i >= 0; --i)
+            {
+                string[] argParts = (originalArgList[i].Substring(1)).Split(':');
+                if (argParts.Length > 0 && String.Equals(argParts[0], "require", StringComparison.OrdinalIgnoreCase))
+                {
+                    originalArgList.RemoveAt(i);
+                }
+            }
```

Comment:
/require:nnn is supported for flexibility, but the hashset approach didn't handle it properly.  I found some instances where the default 1 minute was too short (high load) and did use /require:180 to allow more time.  I'll make the default reasonable, but still want to support this syntax as an emergency out that can be provided by the script.